filament/src/details/Camera.h

```diff
@@ -161,6 +163,7 @@ class FCamera : public Camera {
 
     math::mat4 mProjection;            // projection matrix (infinite far)
     math::mat4 mProjectionForCulling;  // projection matrix (with far plane)
+    math::mat4 mScalingMatrix;         // additional scaling applied to projection
```

Comment:
Store this as a `vec4` representing the diagonal, then everywhere you use `mScalingMatrix` replace with `mat4(mScalingDiagonal)` -- this will actually be a lot more efficient because you're telling the compiler that most coefficient in the matrix are zero (for the multiply), also takes less space.